practices/appeng/adrs/0004-use-testing-library.md

```diff
@@ -0,0 +1,77 @@
+* `-` Another tool to learn
+
+## Impact of a decision (financial, training, transitioning…)
+
+* Potential technical debt of having to convert existing tests to use Testing Library  
+    * This conversion time is an issue, but at root an issue of the technical debt of other libraries, like Enzyme, that aren’t compatible with more recent versions of React, and need to be replaced.  
+* Potential training debt of having to have people learn how to use Testing Library
+   * Also a real issue, but often learning how best to use Testing Library builds transferable knowledge about good markup and the DOM
+
+
+### Use Enzyme
+
+* `-` Enzyme is maintained by one person, and is two major React versions behind.  Effectively defunct.
+* `-` Uses jquery-style queries, which don’t have the benefits of Testing Library’s text- and role-based queries, mentioned above.
+
+
+### Use built in tools in Jest (or other test runner) without adding an additional library 
+
+* `+` Reduces the number of tools developers have to learn
+* `-` Doesn’t adhere to the paradigm of testing in the way the application is actually used
+* `-` The main tool for testing complex rendered code in Jest is a snapshot test, which adds a graphical artifact to the codebase, and immediately breaks and must be updated when the UI changes. This is good for checking for unexpected changes to DOM, but very brittle when adding new features as part of regular development
+* `-`  We don’t lose any Jest functionality by using Testing Library, because Jest is still our test runner.  Mocks, spies, and timers are all still available.
+
+### Don’t test and/or manual testing
+
+(This is not the “test your code” ADR, but for the sake of completeness, here are a few points to keep in mind.  Truss is 100% committed to automated testing.)
+
+* `+` Developers don’t have to write tests.
+* `+` Automated tests require a leap of faith, while manual testing feels more tangible.
+* `+` Can test more edge cases and states with manual testing that may not have been found or written with automated testing
+* `-` Extra time is spent doing manual testing
+* `-` Manual testing is slow and doesn’t scale
+* `-` Much tighter feedback loops with automated testing, which can test a wide range of regressions much more quickly than manual testing.
+* `-` Between Jest and Cypress, which Testing Library works with, it’s very difficult to think of a test scenario that can be manually, but not programmatically, tested.  If it’s important enough to test, very likely it can be automated.
+
+## Circumstances to not use this ADR / Exceptions
+
+There will be projects where existing code has tests already written in other styles, not using Testing Library.  In those cases, the team will have to decide whether the effort to convert is worthwhile.  If something like Enzyme is blocking upgrading other libraries, it may be worthwhile, but in some cases the team might decide that the conversion time is better spent elsewhere.
```

Comment:
QUESTION: `something like Enzyme` is this meant to refer to Enzyme specifically or any framework?